egs/mgb2_arabic/s5/local/mgb_data_prep.sh

```diff
@@ -39,11 +33,21 @@ head -500 $train_dir/wav_list > $train_dir/wav_list.short
 set -e -o pipefail
 
 xmldir=$db_dir/train/xml/bw
-cat $train_dir/wav_list | while read basename; do
+if [ -z $(which xml) ]; then
+  echo "xml not found, using BeautifulSoup in python"
+  cat $train_dir/wav_list | while read basename; do
+    [ ! -e $xmldir/$basename.xml ] && echo "Missing $xmldir/$basename.xml" && exit 1
+    local/process_xml.py $xmldir/$basename.xml - | local/add_to_datadir.py $basename $train_dir $mer
+    echo $basename $wavDir/$basename.wav >> $train_dir/wav.scp
+  done
+else
+  echo "xml found, using it"
+  cat $train_dir/wav_list | while read basename; do
     [ ! -e $xmldir/$basename.xml ] && echo "Missing $xmldir/$basename.xml" && exit 1
     xml sel -t -m '//segments[@annotation_id="transcript_align"]' -m "segment" -n -v  "concat(@who,' ',@starttime,' ',@endtime,' ',@WMER,' ')" -m "element" -v "concat(text(),' ')" $xmldir/$basename.xml | local/add_to_datadir.py $basename $train_dir $mer
     echo $basename $wavDir/$basename.wav >> $train_dir/wav.scp
-done 
+  done
+fi
```

Comment:
Did you check both approaches produce the same outputs?

Reply:
Yes, but in different order

Reply:
> Did you check both approaches produce the same outputs?

Yes, but in different order.